lib/oli/delivery/evaluation/explanation.ex

```diff
@@ -12,9 +13,15 @@ defmodule Oli.Delivery.Evaluation.Explanation do
   """
   def get_explanation(
         %ExplanationContext{
-          part: part
+          part: part,
+          activity_attempt: %ActivityAttempt{} = activity_attempt
         } = context
       ) do
+    context = %{
+      context
+      | activity_attempt: activity_attempt |> Repo.preload(:part_attempts)
```

Comment:
We need to be careful about deeply nesting database preloads like this. Since this function can be called from a higher up Enum iteration, the number of sequential database calls can be `n` here, possibly hammering the database.

At a first glance, it looks like the two areas where these should be preloaded are:
1. `lib/oli/delivery/attempts/activity_lifecycle.ex:134`
2. `lib/oli/delivery/attempts/activity_lifecycle.ex:256`
3. `lib/oli/delivery/attempts/activity_lifecycle/evaluate.ex:716`
4. ...

Changing in `lib/oli/delivery/attempts/core.ex:734` could handle a few of these..